Library/Homebrew/formula_auditor.rb

```diff
@@ -816,6 +816,24 @@ def audit_text
       end
     end
 
+    def audit_github_issue_comment
+      return unless @online
+
+      return unless text.to_s.match?(%r{https://github.com/([-\w_]*)/([-\w_]*)/(pull|issues)/([0-9]*)})
+
+      owner = Regexp.last_match(1)
+      repo = Regexp.last_match(2)
+      id = Regexp.last_match(4)
+
+      # Do not trigger for self references
+      return if repo == formula.tap.repo
+
+      issue = GitHub.API("https://api.github.com/repos/#{owner}/#{repo}/issues/#{id}")
+      return if issue.blank? || JSON.parse(issue)["state"] == "open"
+
+      problem "Formula refers to a GitHub issue or pull request that is closed"
```

Comment:
I think that really depends what we're commenting. If it's a workaround: "use the fix upstream deemed good enough to close the issue", if it's a dependency: "re-evaluate the dependency". 

I mostly see this being useful as a warning for patches that are merged but still apply.

Reply:
> I mostly see this being useful as a warning for patches that are merged but still apply.

Is it worth scoping just to PRs?

Might be worth some sort of nudge on the action but not sure how to make it generic enough?

All this said: fine to merge as-is and iterate.

Reply:
I'm hoping one of the people that do words good will help once this is merged 😄.

> Is it worth scoping just to PRs?

If you mean upstream PRs, I don't think it would be a good trade-off to get better wording instead of a broader scope to catch this issue.

> All this said: fine to merge as-is and iterate.

Planning to write some small test and ensure that it really only catches comments and then I'll merge.